### PR TITLE
Feature/event listener drivers

### PR DESCRIPTION
**Pull request by collindutter**

### CODE REVIEW

File: griptape/utils/stream.py [hunk 1]
```diff
@@ -53,11 +53,14 @@ def run(self, *args) -> Iterator[TextArtifact]:
         t.join()
 
     def _run_structure(self, *args):
+        from griptape.drivers import LocalEventListenerDriver
+
         def event_handler(event: BaseEvent):
             self._event_queue.put(event)
 
         stream_event_listener = EventListener(
-            event_handler, event_types=[CompletionChunkEvent, FinishPromptEvent, FinishStructureRunEvent]
```
Comment (dylanholmes) [inline griptape/utils/stream.py:60]: Now I see why you overrode `publish_event` in `LocalEventListenerDriver`

**Review by dylanholmes — DISMISSED** [review]
🙌

File: griptape/utils/stream.py (hunk 1 above)
Comment (collindutter) [reply griptape/utils/stream.py:60]: 🧵

**Review by andrewfrench — APPROVED** [review]
Very cool.

**Pull request merged**